File: gapic-generator-java/src/main/java/com/google/api/generator/gapic/model/Message.java
// Copyright 2020 Google LLC
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

package com.google.api.generator.gapic.model;

import com.google.auto.value.AutoValue;
import com.google.common.collect.ImmutableList;
import java.util.List;

@AutoValue
public abstract class Message {
  // TODO(miraleung): Get nested messages and enums.

  public abstract String name();

  public abstract ImmutableList<Field> fields();

  public static Builder builder() {
    return new AutoValue_Message.Builder();
  }

  @AutoValue.Builder
  public abstract static class Builder {
    public abstract Builder setName(String name);

    public abstract Builder setFields(List<Field> methods);

    public abstract Message build();
  }
}

File: internal/librariangen/message/message.go
// Copyright 2025 Google LLC
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

// Package message defines data types which the Librarian CLI and language
// containers exchange.
// There shouldn't be CLI-specific data type or language container-specific
// data types in this package.
// TODO(b/447404382): Move this package to the https://github.com/googleapis/librarian
// GitHub repository once the interface is finalized.
// This package must not have any Java-specific implementation.
package message

// GenerateRequest is the JSON message sent to the language container by
// the Librarian CLI when the "generate" command is invoked.
type GenerateRequest struct {
}

// GenerateResponse is the JSON message sent back to the Librarian CLI
// by the language container after processing the "generate" command.
type GenerateResponse struct {
}

// ConfigureRequest is the JSON message sent to the language container by
// the Librarian CLI when the "configure" command is invoked.
type ConfigureRequest struct {
}

// ConfigureResponse is the JSON message sent back to the Librarian CLI
// by the language container after processing the "configure" command.
type ConfigureResponse struct {
}
